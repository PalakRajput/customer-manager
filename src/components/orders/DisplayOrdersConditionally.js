import React from 'react';
import { Link } from 'react-router-dom';
import './index.css'

const DisplayOrdersConditionally = ({customerProp}) => {
    return (
        <>
        <div className='container'>
            <br />
            {customerProp.map(customer => (
                
                <div className='row' style={{ marginLeft: '20%' }} key={customer.id}>
                    
                    {customer.ordersList.length > 0 ?
                        <div>
                            <span className='fullName'>{customer.fname + " " + customer.lname}</span>
                            <table className='table-bordered' >
                                <thead>
                                    <tr>
                                        <th>Product</th>
                                        <th>Quantity</th>
                                        <th>Unit Price</th>
                                        <th>Total</th>
                                    </tr>
                                </thead>
                                <tbody className='table-striped'>
                                    
                                    {customer.ordersList.map(order => (
                                        <tr key={order.oid}>
                                            <td>{order.pname}</td>
                                            <td>{order.quantity}</td>
                                            <td>{order.unitPrice}</td>
                                            <td>${order.quantity * order.unitPrice}</td>
                                        </tr>
                                    ))}
                                    <tr ><td colSpan="4" >${customer.ordersList.reduce((prev, curr) => prev += (curr.quantity*curr.unitPrice),+0)}</td></tr>
                                    {/* {params.id !== undefined ? <tr><a>Add Orders</a></tr> : null} */}
                                </tbody>
                            </table>
                            {/* {customerProp.length === 0 ? <Link to={`/addOrder/${customerProp[0].id}`} role='button' className='btn btn-primary btn-sm' id='orderButton' >Add Orders for {customerProp[0].fname+ " " +customerProp[0].lname}</Link> : null} */}
                            <br /><br />
                        </div> :  (customerProp.length === 1 ? <Link to={`/addOrder/${customerProp[0].id}`} role='button' className='btn btn-primary btn-sm' id='orderButton' >Add Orders for {customerProp[0].fname+ " " +customerProp[0].lname}</Link> : null)
}
                </div>
            ))}


        </div>

    </>
    )
}

export default DisplayOrdersConditionally;