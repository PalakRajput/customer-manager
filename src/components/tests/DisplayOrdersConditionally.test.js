import { render, screen } from '@testing-library/react'
import React from 'react'
import DisplayOrdersConditionally from '../orders/DisplayOrdersConditionally';

describe("Display Customers", () => {
    it (" should render correctly ", () => {
        const customerProps = [
            {id:1, fname:"John", lname:"Doe", city:"LA", ordersList:[
                {oid:1, pname:"Speaker", quantity:1, unitPrice: 20},
                {oid:2, pname:"Bluetooth Device", quantity:1, unitPrice: 10}
            ]}
        ]
           const {asFragment} =  render(<DisplayOrdersConditionally customerProp={customerProps}/>)
           expect(asFragment(<DisplayOrdersConditionally />)).toMatchSnapshot();
    })

})