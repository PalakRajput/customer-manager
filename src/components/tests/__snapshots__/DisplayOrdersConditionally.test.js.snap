// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display Customers  should render correctly  1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <br />
    <div
      class="row"
      style="margin-left: 20%;"
    >
      <div>
        <span
          class="fullName"
        >
          John Doe
        </span>
        <table
          class="table-bordered"
        >
          <thead>
            <tr>
              <th>
                Product
              </th>
              <th>
                Quantity
              </th>
              <th>
                Unit Price
              </th>
              <th>
                Total
              </th>
            </tr>
          </thead>
          <tbody
            class="table-striped"
          >
            <tr>
              <td>
                Speaker
              </td>
              <td>
                1
              </td>
              <td>
                20
              </td>
              <td>
                $20
              </td>
            </tr>
            <tr>
              <td>
                Bluetooth Device
              </td>
              <td>
                1
              </td>
              <td>
                10
              </td>
              <td>
                $10
              </td>
            </tr>
            <tr>
              <td
                colspan="4"
              >
                $30
              </td>
            </tr>
          </tbody>
        </table>
        <br />
        <br />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
